modules/javafx.graphics/src/main/java/com/sun/javafx/font/PrismFontFile.java

```diff
@@ -1217,7 +1237,7 @@ public float getAdvance(int glyphCode, float ptSize) {
             } else {
                 cadv = advanceWidths[numHMetrics-1];
             }
-            return ((float)(cadv & 0xffff)*ptSize)/upem;
+            return ((cadv & 0xffff)*ptSize)/upem;
```

Comment:
This is an example where I think the explicit cast makes it easier to quickly tell at a glance that it is correct, even though it happens to be unnecessary. In this case, I would need to look at the definition of `ptSize` and `upem` to know that it is unneeded, and thus correct without it.

Reply:
Eclipse knows it is unnecessary.  Personally, I am in favor of removing such unnecessary code.

Reply:
If it aids readability, I don't see it as unnecessary. In this case, since the types of all of the variables are not known, you could certainly argue it doesn't matter much -- you need to look at the types to reason about what it does.

Reply:
I don't see it as a useful warning. The code isn't necessarily better without, and can be less clear. I think this might argue for asking Eclipse users to disable this warning in the IDE.

Reply:
I do agree with @kevinrushforth here.

It might be borderline useful to enable the warning once just to see if there are any bugs or perhaps to clean up the most obvious cases, but it probably makes little sense to enable the said warning for large and long lived code bases like this one.

Reply:
I wrote what I think in [this comment](https://github.com/openjdk/jfx/pull/960#issuecomment-1325782856). Even if in this specific case the cast should not be reverted, the warning is very useful to have since it encapsulates non-numeric types as well.

Reply:
@kevinrushforth Is this okay as it is?  Even though the benefit of removing the cast here is minimal, removing it still seems better as it won't need to be suppressed later if we decide to enforce the rules as part of the build.  In this specific case, there was no obvious way to "break" the code (by losing precision) by changing something unrelated (like changing `ptSize` to `double` or `long`).

Reply:
Yes, this is OK.